src/array.cr

```diff
@@ -1823,6 +1823,12 @@ class Array(T)
     self
   end
 
+  # Prints a nicely readable and concise string representation of this array
+  # to *io*.
+  #
+  # The result resembles an array literal but it does not necessarily compile.
+  #
+  # Each element is presented using its `#inspect(io)` result to avoid ambiguity.
```

Comment:
Should we mention avoiding circular references here? Or perhaps some other wording that explains the `...`?  The docs also serve for people writing overrides and I think catching circular references is important for those overriders to know about.

Reply:
I think we only have to revisit this if/when we consider making `exec_recursive` a public API